src/Microsoft.AspNet.Mvc.ModelBinding/Metadata/AssociatedMetadataProvider.cs

```diff
@@ -18,13 +18,12 @@ public abstract class AssociatedMetadataProvider<TModelMetadata> : IModelMetadat
         private readonly ConcurrentDictionary<Type, Dictionary<string, PropertyInformation>> _typePropertyInfoCache =
                 new ConcurrentDictionary<Type, Dictionary<string, PropertyInformation>>();
 
-        public IEnumerable<ModelMetadata> GetMetadataForProperties(object container, [NotNull] Type containerType)
+        public IEnumerable<ModelMetadata> GetMetadataForProperties([NotNull] Type containerType)
         {
-            return GetMetadataForPropertiesCore(container, containerType);
+            return GetMetadataForPropertiesCore(containerType);
         }
 
-        public ModelMetadata GetMetadataForProperty(Func<object> modelAccessor,
-                                                    [NotNull] Type containerType,
+        public ModelMetadata GetMetadataForProperty([NotNull] Type containerType,
                                                     [NotNull] string propertyName)
```

Comment:
agree previous ordering was odd but am not sure we should change this prior to a general API review